aws/resource_aws_autoscaling_policy.go

```diff
@@ -71,6 +61,137 @@ func resourceAwsAutoscalingPolicy() *schema.Resource {
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"predefined_metric_pair_specification": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"predefined_metric_type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGCPUUtilization",
+														"ASGNetworkIn",
+														"ASGNetworkOut",
+														"ALBRequestCount",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"predefined_scaling_metric_specification": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"predefined_metric_type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGAverageCPUUtilization",
+														"ASGAverageNetworkIn",
+														"ASGAverageNetworkOut",
+														"ALBRequestCountPerTarget",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"predefined_load_metric_specification": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"predefined_metric_type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ASGTotalCPUUtilization",
+														"ASGTotalNetworkIn",
+														"ASGTotalNetworkOut",
+														"ALBTargetGroupRequestCount",
+													}, false),
+												},
+												"resource_label": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+									"target_value": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"max_capacity_breach_behavior": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "HonorMaxCapacity",
+							ValidateFunc: validation.StringInSlice([]string{
+								"HonorMaxCapacity",
+								"IncreaseMaxCapacity",
+							}, false),
+						},
+						"max_capacity_buffer": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      0,
+							ValidateFunc: validation.IntBetween(-1, 100),
```

Comment:
The documentation says the range is from 0-100, and that if set to 0 "may scale capacity higher than the maximum capacity to equal but not exceed forecast capacity", instead of having that behaviour at -1.

If we need to distinguish between `0` and unset, we should use a `nullable.Int` https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/internal/experimental/nullable/int.go

There should probably be an acceptance test to validate `0` vs unset